assets/hip-1056/protobuf/stream/output/smart_contract_service.proto

```diff
@@ -0,0 +1,183 @@
+/**
+ * #  Service
+ * Block stream messages that report the results of transactions handled
+ * by the `smart contract` service.
+ *
+ * > REVIEW NOTE
+ * >> The use of sidecar records is a bit odd here. We may find it more
+ * >> effective to extract the actual changes into proper output messages
+ * >> and fields included in the ethereum call output and/or related state
+ * >> changes items, and remove the whole sidecar concept going forward.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in
+ * [RFC2119](https://www.ietf.org/rfc/rfc2119) and clarified in
+ * [RFC8174](https://www.ietf.org/rfc/rfc8174).
+ */
+syntax = "proto3";
+
+package com.hedera.hapi.block.stream.output;
+
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.block.stream.output.protoc";
+// <<<pbj.java_package = "com.hedera.hapi.block.stream.output">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "contract_call_local.proto";
+import "sidecar_file.proto";
+
+/**
+ * Block Stream data for a `contractCallMethod` transaction.
+ *
+ * This message SHALL NOT duplicate information already contained in
+ * the original transaction.
+ */
+message CallContractOutput {
+     /**
+     * A result for a Contract Call Transaction execution, either a create or call.
+     * <p>
+     * This field SHALL contain all of the data produced by the contract
+     * create transaction as well as basic accounting results.
+     */
+     proto.ContractFunctionResult contract_function_result = 1;
+ 
+     /**
+     * The storage changes to a smart contract's storage as a side effect of the function call.
+     */
+     repeated ContractStateChange state_changes = 2;
```

Comment:
A little concerned about this item.  Changes to contract storage _must_ be present in the state changes; so this _should_ always be duplicate data.

Reply:
Great.
To confirm `MapChangeValue` will contain the state changes for each slot?
[ContractStateChange](https://github.com/hashgraph/hedera-protobufs/blob/main/streams/contract_state_change.proto) also captured which contract was being changed, the value read and written. Can we still obtain both?
Agreed on not wanting duplicate data so happy to scope this down even more

Reply:
In state changes, the `SlotKey` includes both contract ID and a `bytes` key.
The SlotValue contains three bytes fields (value, previous key, and next key).
Each change to a contract slot _should_ produce a separate `StateChange` with the relevant slot key and slot value.
The changes for a transaction must follow that transaction in the stream, so they can be tied together via block stream order.

Reply:
The value before is necessary, thats what the read value was capturing. This is for EVM equivalence

Reply:
Great, is slotkey also includes ContractID then i think we could remove this 😄

Reply:
`SlotValue` actually has everything we need so we should be good. I'll remove it

Reply:
Removed